dubbo-monitor/dubbo-monitor-default/src/main/java/org/apache/dubbo/monitor/dubbo/MetricsFilter.java

```diff
@@ -0,0 +1,251 @@
+            ExecutorService executor = (ExecutorService) entry.getValue();
+            if (executor instanceof ThreadPoolExecutor) {
+                ThreadPoolExecutor tp = (ThreadPoolExecutor) executor;
+                // ignore metrcis service
+                if (port.equals(this.port + "")) {
+                    continue;
+                }
+
+                threadPoolMtricList.add(value2MetricObject("threadPool.active", tp.getActiveCount(), MetricLevel.MAJOR));
+                threadPoolMtricList.add(value2MetricObject("threadPool.core", tp.getCorePoolSize(), MetricLevel.MAJOR));
+                threadPoolMtricList.add(value2MetricObject("threadPool.max", tp.getMaximumPoolSize(), MetricLevel.MAJOR));
+                threadPoolMtricList.add(value2MetricObject("threadPool.current", tp.getPoolSize(), MetricLevel.MAJOR));
+            }
+        }
+
+        return threadPoolMtricList;
+    }
+
+    private MetricObject value2MetricObject(String metric, Integer value, MetricLevel level) {
+        if (metric == null || value == null || level == null)
+            return null;
+
+        return new MetricObject
+                .Builder(metric)
+                .withValue(value)
+                .withLevel(level)
+                .build();
+    }
+
+    private void initMetricsInvoker() {
+        metricsInvoker = new Invoker<MetricsService>() {
+            @Override
+            public Class<MetricsService> getInterface() {
+                return MetricsService.class;
+            }
+
+            @Override
+            public Result invoke(Invocation invocation) throws RpcException {
+                String group = invocation.getArguments()[0].toString();
+                MetricRegistry registry = MetricManager.getIMetricManager().getMetricRegistryByGroup(group);
+
+                SortedMap<MetricName, FastCompass> fastCompasses = registry.getFastCompasses();
+
+                long timestamp = System.currentTimeMillis();
+                double rateFactor = TimeUnit.SECONDS.toSeconds(1);
+                double durationFactor = 1.0 / TimeUnit.MILLISECONDS.toNanos(1);
+
+
+                MetricsCollector collector = MetricsCollectorFactory.createNew(
+                        CollectLevel.NORMAL, Collections.EMPTY_MAP, rateFactor, durationFactor, null);
+
+                for (Map.Entry<MetricName, FastCompass> entry : fastCompasses.entrySet()) {
+                    collector.collect(entry.getKey(), entry.getValue(), timestamp);
+                }
+
+                RpcResult result = new RpcResult();
+
+                List res = collector.build();
+                res.addAll(getThreadPoolMessage());
+                result.setValue(JSON.toJSONString(res));
+                return result;
+            }
+
+            @Override
+            public URL getUrl() {
+                return URL.valueOf(protocolName + "://" + NetUtils.getLocalAddress().getHostName() + ":" + port + "/" + MetricsService.class.getName());
```

Comment:
`NetUtils.getLocalAddress().getHostName()`

The IP address used here may have a problem, we should think using a unified tool for getting outward IP, I remember there's one thread discussing the same topic.